datalad/distribution/tests/test_install.py

```diff
@@ -570,6 +570,33 @@ def test_install_list_arguments(src, path, path_outside):
         for skipped in ['not_existing', path_outside]:
             cml.assert_logged(msg="Installation of {0} skipped".format(skipped),
                               regex=False, level='INFO')
+        ok_(isinstance(result, list))
+        eq_(len(result), 2)
         for sub in [Dataset(opj(path, 'subm 1')), Dataset(opj(path, 'subm 2'))]:
             assert_in(sub, result)
             ok_(sub.is_installed())
+
+    # install list and have single result item:
+    result = ds.install(path=['subm 1', 'not_existing'])
+    eq_(result, Dataset(opj(path, 'subm 1')))
+
+
+@with_testrepos('submodule_annex', flavors=['local'])
+@with_tempfile(mkdir=True)
+def test_install_skip_failed_recursive(src, path):
+
+    # install top level:
+    ds = install(path=path, source=src)
+    sub1 = Dataset(opj(path, 'subm 1'))
+    sub2 = Dataset(opj(path, 'subm 2'))
+    # sabotage recursive installation of 'subm 1' by polluting the target:
+    with open(opj(path, 'subm 1', 'blocking.txt'), "w")  as f:
+        f.write("sdfdsf")
```

Comment:
Just FYI I usually use `create_tree` (parent of `@with_tree`) when I need to create some files/dirs for testing